big_scape/comparison/extend.py

```diff
@@ -592,3 +592,107 @@ def extend_greedy(pair: RecordPair) -> None:
 
     logging.debug("after greedy extend:")
     logging.debug(pair.comparable_region)
+
+
+def extend_simple_match(pair: RecordPair, match, gap):
+    """Performs extension by first creating a simple match matrix, then
+    performing a match/gap extentsion similar to legaxy expansion
+
+    This method expects LCS to have been performed on the pair, and will
+    do all four directions at once
+
+    Args:
+        pair: The record pair to extend
+        match: The score for a match
+        gap: The score for a gap
+    """
+
+    # I *think* the simplest way to do this is to create a list of domains with their source cds. for reasons.
+    a_domains = []
+    b_domains = []
+
+    # so we'll do a loop through cds and through domains to keep track of everything
+    for cds_idx, cds in enumerate(pair.record_a.get_cds()):
+        for domain in cds.hsps:
+            a_domains.append((domain, cds_idx))
+
+    for cds_idx, cds in enumerate(pair.record_b.get_cds()):
+        for domain in cds.hsps:
+            b_domains.append((domain, cds_idx))
+
+    # get the common domains
+    common_domains = set([a[0] for a in a_domains]).intersection(
+        set([b[0] for b in b_domains])
+    )
+
+    # TODO: this is obviously repetitive, but beyond extracting a function I'm
+    # not sure how to make it better
+    # for now I'd prefer this to be obvious and clear over looking good
+
+    # a forward
+    score = 0
+    max_score = 0
+    for i in range(pair.comparable_region.lcs_a_stop + 1, len(a_domains)):
```

Comment:
lcs_a_stop governs cds bounds, should be lcs_domain_a_stop.
comparable region stops are exclusive, so remove the +1
```suggestion
    for i in range(pair.comparable_region.lcs_domain_a_stop, len(a_domains)):
```